pubsub/azurepubsub/example_test.go

```diff
@@ -117,3 +121,80 @@ func ExampleOpenSubscription() {
 	// See https://godoc.org/github.com/Azure/azure-service-bus-go#Message.Complete.
 	msg.Ack()
 }
+
+func ExampleSubscriptionWithAutoDeleteAndNoAck() {
+
+	ctx := context.Background()
+	// See docs below on how to provision an Azure Service Bus Namespace and obtaining the connection string.
+	// https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-dotnet-get-started-with-queues
+	connString := os.Getenv("SERVICEBUS_CONNECTION_STRING")
+	topicName := "test-topic"
+	subscriptionName := "test-sub"
+
+	if connString == "" {
+		log.Fatal("Service Bus ConnectionString is not set")
+	}
+
+	// Construct a Service Bus Namespace from a SAS Token.
+	// See https://godoc.org/github.com/Azure/azure-service-bus-go#Namespace.
+	ns, err := azurepubsub.NewNamespaceFromConnectionString(connString)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Construct a Service Bus Topic for a topicName associated with a NameSpace.
+	// See https://godoc.org/github.com/Azure/azure-service-bus-go#Topic.
+	sbTopic, err := azurepubsub.NewTopic(ns, topicName, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer sbTopic.Close(ctx)
+
+	// Construct Receiver to AutoDelete messages.
+	// See https://godoc.org/github.com/Azure/azure-service-bus-go#SubscriptionWithReceiveAndDelete.
+	var opts []servicebus.SubscriptionOption
+	opts = append(opts, servicebus.SubscriptionWithReceiveAndDelete())
+
+	// Construct a Service Bus Subscription which is a child to a Service Bus Topic.
+	// See https://godoc.org/github.com/Azure/azure-service-bus-go#Topic.NewSubscription.
+	sbSub, err := azurepubsub.NewSubscription(sbTopic, subscriptionName, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer sbSub.Close(ctx)
+
+	// Construct a custom Ack Handler to override the driver acking.
+	// This is useful for a number of reasons including noAcks or custom acks.
+	subOpts := &azurepubsub.SubscriptionOptions{
+		AckOverride: func(ctx context.Context, ids []driver.AckID) error {
```

Comment:
Supply a custom Ack handler to intercept & override the driver ack

Reply:
Consider adding a comment explaining the options you might want here, including panic, log, or no-op.

Reply:
@jba 	maybe we should have something about this in the portable type docs that drivers can point to....?

Reply:
https://github.com/google/go-cloud/pull/1545

Reply:
Ok, I'll use the noted comment as a starter.